homeassistant/components/soundtouch/media_player.py

```diff
@@ -194,6 +205,22 @@ def __init__(self, name, config):
         self._status = self._device.status()
         self._volume = self._device.volume()
         self._config = config
+        self._resources = config.get("resources")
+        self._sources = []
+        self._sourcename = {}
+        self._namesource = {}
+        _LOGGER.debug(f"CONF_RESOURCES: {self._resources}")
+        if self._resources:
+            for pair in self._resources:
+                source = pair["source"].upper()
+                name = pair["name"]
+                self._sources.append(name)
+                self._sourcename[source] = name
+                self._namesource[name] = source
+                self._namesource[source] = source
```

Comment:
What if name and source are the same? Should we guard for that?

Reply:
done.